rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/server/nodesource/NSCreationServlet.java

```diff
@@ -132,6 +132,36 @@ private void createNs(HttpServletRequest request, HttpServletResponse response)
                         populateOrderMap(policyParamOrder, formFieldValue);
                     } else if (formFieldName.equals("policyParamFileOrder")) {
                         populateOrderMap(policyParamFileOrder, formFieldValue);
+                    } else if (formFieldName.equals("hidden-infra") && !formFieldValue.isEmpty()) {
+                        String[] pairs = formFieldValue.split("\\^");
+                        for (String pair : pairs) {
+                            String[] split = pair.split("->");
```

Comment:
To improve: change split and pair to more significant variable names

Reply:
whole this approach (-> and ^) should be changed.
however i want to merge and start to test. After I will discuss it with Fabien I will refactor & improve it.